cypress/README.md

```diff
@@ -0,0 +1,27 @@
+# Cypress Integration Tests
+
+These are end to end tests for kiali that are meant to be run against a live environment.
+
+## Prerequisites
+
+Ensure the `baseUrl` field in the `cypress.json` file at the kiali-ui root is pointing to the server you are trying to test. By default this is `localhost:3000` so if you are running kiali locally you should just need to do `yarn start` before running cypress.
+
+## Running tests
+
+Tests can be run with the cypress browser:
+
+```bash
+yarn cypress
+```
+
+or in headless mode:
+
+```bash
+yarn cypress:run
+```
+
+## Testing Strategies
+
+### Graph
+
+The kiali graph is primarily a canvas element which makes it more difficult to test through query selectors with cypress. To get around this limitation, a simple way of testing visual elements is to do snapshot testing aka [visual testing](https://docs.cypress.io/guides/tooling/visual-testing). This boils down to comparing an existing image of the page to a new image. If any differences exist, the test will fail. There are other ways of testing canvas elements in Cypress but they seem to involve interacting with cypress through a custom api rather than visually which is not ideal.
```

Comment:
Mmm, it's going to be a limitation if cypress can't interact with the canvas element.
I don't think we can't rely on snapshots comparison in the graph, for multiple reasons, like the same data can render different graphs easily due to the nature of the framework/technology we use.

Reply:
But I'd guess it's an early comment from my side, worth giving a try to see how it works and which level of sensitivity of "snapshot-image-comparison" may offer here.

Reply:
It's definitely a limitation but more of a limitation of using html canvas than cypress itself I'd say. There's some settings to tune the diff sensitivity between two images. It'd be good to play around with those to make sure it'll work for two graphs that are rendered slightly differently but are mostly the same.

Reply:
If we want to invest and research visual testing (i.e. that canvas renders), I highly recommend this talk (timestamp 8:30) https://youtu.be/Dl_XMd_1F6E?list=PLI1o8Ea2Lj01qMzZRg8F4tRpIxTklrnbV&t=511
 
^ that pretty much sums up the possible approaches, also he has a brief demo at the end

my opinion: We might want to try create our snapshots, see if that works. If not, some 3rd party commercial service with OSS plan could